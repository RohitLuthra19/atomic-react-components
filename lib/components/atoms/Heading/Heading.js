// @flow
import type { Node, React } from 'react';
import styled from 'styled-components';

import styles from './Heading.style';

type Props = {
  children: Node,
  className: string,
  HeadingType?: string,
  inheritedStyles?: string,
};

const Heading = ({
  children, className, HeadingType, inheritedStyles, ...others
}: Props): Node => (
  <HeadingType className={className} {...others}>
    {children}
  </HeadingType>
);

Heading.defaultProps = {
  inheritedStyles: '',
  HeadingType: 'h1',
};

export default styled(Heading)`${styles}`;
export { Heading as HeadingVanilla };
