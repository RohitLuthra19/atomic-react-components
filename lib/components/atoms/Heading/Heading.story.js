// @flow
import React from 'react';
import { storiesOf } from '@storybook/react';

import mainHeading from './Heading.mock';
// Import Vanilla Component to display markup and props
import { HeadingVanilla } from './Heading';
// Import Styled Component to showcase variations
import Heading from './index';

storiesOf('Atoms', module).addWithChapters('Heading', {
  chapters: [
    {
      sections: [
        {
          sectionFn: () => <HeadingVanilla {...mainHeading} className="hide-default-sample">This is Heading text</HeadingVanilla>,
          options: {
            showSource: true,
            allowSourceToggling: true,
            showPropTables: true,
            allowPropTablesToggling: true,
          },
        },
      ],
    },
    {
      title: 'Input Field Error Message',
      sections: [
        {
          title: 'Default',
          sectionFn: () => <Heading {...mainHeading}>This is Heading text</Heading>,
        },
      ],
    },
  ],
});
