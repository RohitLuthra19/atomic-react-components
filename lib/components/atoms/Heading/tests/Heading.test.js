import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import Heading from '..';

const props = {
  children: 'Sample',
  className: '',
};

describe('Heading Component', () => {
  let heading;

  test('should render correctly and match snapshot', () => {
    heading = shallow(<Heading {...props} />);
    expect(heading).toMatchSnapshot();
  });
});
