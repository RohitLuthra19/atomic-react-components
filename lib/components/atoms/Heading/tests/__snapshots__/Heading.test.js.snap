// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading Component should render correctly and match snapshot 1`] = `
.c0 {
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-weight: normal;
  font-size: 20px;
}

<Heading
  HeadingType="h1"
  className="c0"
  elementLocator=""
  inheritedStyles=""
>
  Sample
</Heading>
`;
